lib/Frontend/XCTestMethodsEmitter.cpp

```diff
@@ -0,0 +1,135 @@
+//===--- ParseableInterfaceSupport.cpp - swiftinterface files ------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Frontend/XCTestMethodsEmitter.h"
+#include "swift/Basic/JSONSerialization.h"
+#include "swift/AST/ASTContext.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+
+using namespace swift;
+
+// Copied from lib/Index/IndexSymbol.cpp
```

Comment:
These 3 functions are copied from IndexSymbol.cpp. Perhaps, we should put them in a shared location?

Reply:
Yes, please factor them out. I don't know if Frontend depends on Index; if not you can move them to IDE.

Reply:
Definitely. I don't remember if Frontend depends on Index; if not, it very likely depends on IDE.